src/data/equiv/module.lean

```diff
@@ -477,3 +477,59 @@ def to_module_aut : S →* M ≃ₗ[R] M :=
   map_mul' := λ a b, linear_equiv.ext $ mul_smul _ _ }
 
 end distrib_mul_action
+
+namespace add_equiv
+variables [semiring R] [add_comm_monoid M] [module R M] [add_comm_monoid M₂] [module R M₂]
+
+/-- An additive equivalence whose underlying function preserves `smul` is a linear equivalence. -/
+def to_linear_equiv (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) : M ≃ₗ[R] M₂ :=
+{ map_smul' := h, .. e, }
+
+@[simp] lemma coe_to_linear_equiv (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
+  ⇑(e.to_linear_equiv h) = e :=
+rfl
+
+@[simp] lemma coe_to_linear_equiv_symm (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
+  ⇑(e.to_linear_equiv h).symm = e.symm :=
+rfl
+
+/-- An additive equivalence between commutative groups is a linear equivalence between ℤ-modules -/
+def to_int_linear_equiv {α β : Type*} [add_comm_group α] [add_comm_group β]
+  (e : α ≃+ β) : α ≃ₗ[ℤ] β :=
+e.to_linear_equiv $ λ c a, by { erw e.to_add_monoid_hom.map_zsmul, refl }
+
+@[simp] lemma coe_to_int_linear_equiv {α β : Type*} [add_comm_group α] [add_comm_group β]
+  (e : α ≃+ β) : ⇑(e.to_int_linear_equiv) = e := rfl
+
+@[simp] lemma coe_to_int_linear_equiv_symm {α β : Type*} [add_comm_group α] [add_comm_group β]
+  (e : α ≃+ β) : ⇑(e.to_int_linear_equiv).symm = e.symm := rfl
+
+@[simp] lemma to_int_linear_equiv_refl {α : Type*} [add_comm_group α]:
+  ((add_equiv.refl α).to_int_linear_equiv) = linear_equiv.refl ℤ α := rfl
+
+@[simp] lemma to_int_linear_equiv_trans
+  {α β γ : Type*} [add_comm_group α] [add_comm_group β] [add_comm_group γ]
+  (e₁ : α ≃+ β) (e₂ : β ≃+ γ) :
+  (e₁.to_int_linear_equiv).trans (e₂.to_int_linear_equiv) = (e₁.trans e₂).to_int_linear_equiv :=
+rfl
+
+/-- An additive equivalence between commutative monoids is a linear equivalence between ℕ-modules -/
+def to_nat_linear_equiv {α β : Type*} [add_comm_monoid α] [add_comm_monoid β]
+  (e : α ≃+ β) : α ≃ₗ[ℕ] β :=
+e.to_linear_equiv $ λ c a, by { erw e.to_add_monoid_hom.map_nsmul, refl }
+
+@[simp] lemma coe_to_nat_linear_equiv {α β : Type*} [add_comm_monoid α] [add_comm_monoid β]
+  (e : α ≃+ β) : ⇑(e.to_nat_linear_equiv) = e := rfl
+
+@[simp] lemma coe_to_nat_linear_equiv_symm {α β : Type*} [add_comm_monoid α] [add_comm_monoid β]
+  (e : α ≃+ β) : ⇑(e.to_nat_linear_equiv).symm = e.symm := rfl
```

Comment:
A stronger statement is
```suggestion
  (e : α ≃+ β) : (e.to_nat_linear_equiv).symm = e.symm.to_nat_linear_equiv := rfl
```

Reply:
Of course! shouldn’t have cargo-culted that from above.

BTW, should I try to change `coe_to_linear_equiv_symm` too to something like
```
@[simp] lemma coe_to_linear_equiv_symm (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
  (e.to_linear_equiv h).symm = e.symm.to_linear_equiv (λ c x, TODO) :=
rfl
```

Reply:
But maybe in a separate PR, in case it breaks something downstream, and to make progress here :-)

Reply:
I think probably best to declare it out of scope for this PR.

Reply:
Ha, I came up with this nice lemma, but it looks like a bad idea, as it might make the simplifier loop :)
```
@[simp] lemma coe_to_linear_equiv_symm (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
  (e.to_linear_equiv h).symm = e.symm.to_linear_equiv ((e.to_linear_equiv h).symm.map_smul') :=
rfl
```
I’ll ignore this distraction for now.